### PR TITLE
feat: add thanos for prom HA

### PR DESCRIPTION
**Pull request by srodenhuis**
This PR adds Thanos for Prometheus HA and includes:
- add Thanos Helm chart
- configure Prometheus to use Thanos sidecars
- create `objstore.yml` for OBJ configuration based on configured OBJ provider
- add thanos-query datasource to all Grafana's
- add Thanos Grafana dashboards to Platform Grafana

## Testing

- Configure OBJ with a bucket for thanos
- Activate Thanos app
- Check to see if the thanos sidecar is added to the Prometheus pod
- Check (using the Thanos query UI) to see if the sidecar endpoint status is up
- Check if (this can take some time) if the Store endpoint (only one) has status Up
- Set the replica's for prometheus to `2`
- Check to see if the second sidecar endpoint comes up
- Check to see if all the announced LabelSets (for both replica's) are shown

You should now see the following:

Screenshot 2024-09-25 at 14 41 17

## Checklist

- [x] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [x] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [x] Helm releases are meeting otomi's baseline security policies, if applicable.

### CODE REVIEW

**Review by j-zimnowoda — COMMENTED** [review]
I noticed that network policies are deployed once Thanos is deployed. I am pretty sure that they are compatible with Thanos. 
Just a reminder how netpol works:
If there is not network policy then nothing gets blocked. The moment you deploy a network policy on traffic that is defined in netpol will be allow. 

Keeping the above in mind, 
1) is there any impact on other apps that are deployed to the monitoring space? 
2) Would you consider deploying Thanos to its own namespace, so we can mitigate such conflicts?

**Comment by Ani1357** [comment]
### Test results: 
- [x] Configure OBJ with a bucket for thanos: ✅
- [x] Activate Thanos app: ✅
- [x] Check to see if the thanos sidecar is added to the Prometheus pod: ✅
- [x] Check (using the Thanos query UI) to see if the sidecar endpoint status is up: ✅
- [x] Check if (this can take some time) if the Store endpoint (only one) has status Up: ✅
- [x] Set the replica's for prometheus to 2: ✅
- [x] Check to see if the second sidecar endpoint comes up: ✅
- [ ] Check to see if all the announced LabelSets (for both replica's) are shown: 🚫 After 24hours I can see only one Announced LableSet
image

File: charts/thanos/templates/storegateway/networkpolicy.yaml
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply charts/thanos/templates/storegateway/networkpolicy.yaml:1]: Thanos now in its own namespace

**Review by j-zimnowoda — APPROVED** [review]

**Review by Ani1357 — APPROVED** [review]
Tested again. All tests were successful:
image

**Pull request merged**